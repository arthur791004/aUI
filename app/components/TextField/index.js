import React, { PureComponent } from 'react';
import styled from 'styled-components';
import { timingFunctions } from 'polished';

const TextFieldWrapper = styled.div`
  position: relative;
  font-size: 16px;
  line-height: 24px;
  width: 300px;
  height: 66px;
`;

const Label = styled.label`
  position: absolute;
  line-height: 24px;
  top: 20px;
  z-index: 1;
  transform:
    scale(${props => (props.isFocused ? 0.75: 1)})
    translate(0px, ${props => (props.isFocused ? -26 : 0)}px);
  transform-origin: left top 0px;
  pointer-events: none;
  user-select: none;
  transition: all 0.45s ${timingFunctions('easeOutQuint')};
  color: ${props => (props.isFocused ? '#28232D' : '#A9A4B0')};
`;

const Input = styled.input`
  position: relative;
  top: 20px;
  padding: 0px;
  width: 100%;
  border: none;
  outline: none;
  color: #28232D;
  box-sizing: border-box;
  appearance: textfield;
`;

const UnderlineInactive = styled.div`
  position: absolute;
  border-bottom: 1px solid #D2D2D7;
  bottom: 21px;
  box-sizing: content-box;
  margin: 0px auto;
  width: 100%;
`;

const UnderlineActive = UnderlineInactive.extend`
  border-bottom: 1px solid #28232D;
  left: 0;
  right: 0;
  width: ${props => (props.isFocused ? '100%' : 0)};
  transition: width 0.5s ${timingFunctions('easeOutQuint')};
`;

class TextField extends PureComponent {
  state = {
    isFocused: false,
  }

  onFocus = () => {
    this.setState({ isFocused: true });
  }

  onBlur = () => {
    this.setState({ isFocused: false });
  }

  render() {
    const { label } = this.props;
    const { isFocused } = this.state;

    return (
      <TextFieldWrapper>
        <Label isFocused={isFocused}>
          {label}
        </Label>
        <Input
          onFocus={this.onFocus}
          onBlur={this.onBlur}
        />
        <UnderlineInactive />
        <UnderlineActive isFocused={isFocused} />
      </TextFieldWrapper>
    );
  }
}

export default TextField;
