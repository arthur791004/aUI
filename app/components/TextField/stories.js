import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import TextField from './';

storiesOf('TextField', module)
  .add('basic', () => (
    <TextField label="username" />
  ));
