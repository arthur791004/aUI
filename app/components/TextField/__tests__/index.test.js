import React from 'react';
import { shallow } from 'enzyme';

import TextField from '../';

describe('<TextField />', () => {
  it('should render the component', () => {
    const wrapper = shallow(
      <TextField label="label" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
