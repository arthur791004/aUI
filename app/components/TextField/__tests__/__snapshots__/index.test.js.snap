// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render the component 1`] = `
<styled.div>
  <styled.label
    isFocused={false}
  >
    label
  </styled.label>
  <styled.input
    onBlur={[Function]}
    onFocus={[Function]}
  />
  <styled.div />
  <styled.div
    isFocused={false}
  />
</styled.div>
`;
